operator/src/main/resources/scripts/wdt_mii_filter.py

```diff
@@ -0,0 +1,546 @@
+# Copyright (c) 2018, 2021, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+#
+# ------------
+# Description:
+# ------------
+#  This is a WDT filter for handling WLS configuration overrides.
+#
+
+import inspect
+import os
+import sys
+
+tmp_callerframerecord = inspect.stack()[0]    # 0 represents this line # 1 represents line at caller
+tmp_info = inspect.getframeinfo(tmp_callerframerecord[0])
+tmp_scriptdir=os.path.dirname(tmp_info[0])
+sys.path.append(tmp_scriptdir)
+
+env = None
+
+class OfflineWlstEnv(object):
+
+  def open(self, model):
+
+    self.model = model
+    # before doing anything, get each env var and verify it exists
+
+    self.DOMAIN_UID               = self.getEnv('DOMAIN_UID')
+    self.DOMAIN_HOME              = self.getEnv('DOMAIN_HOME')
+    self.LOG_HOME                 = self.getEnv('LOG_HOME')
+    self.ACCESS_LOG_IN_LOG_HOME   = self.getEnvOrDef('ACCESS_LOG_IN_LOG_HOME', 'true')
+    self.DATA_HOME                = self.getEnvOrDef('DATA_HOME', "")
+    self.CREDENTIALS_SECRET_NAME  = self.getEnv('CREDENTIALS_SECRET_NAME')
+
+    # initialize globals
+    self.INTROSPECT_HOME          = '/tmp/introspect/' + self.DOMAIN_UID
+    self.CREDENTIALS_SECRET_PATH = self.getEnvOrDef('CREDENTIALS_SECRET_PATH', '/weblogic-operator/secrets')
+
+    if model and 'topology' in model:
+      self.DOMAIN_NAME = model['topology']['Name']
```

Comment:
I assume we always expects domain-name for the MII case?  If so, maybe have this script error out if it's not found...

Reply:
For online updates, I noticed that WDT updateDomain command is processing a sparse model fragment where the topology doesn't specify the 'Name' attribute (domain-name).  The only configuration override that this is verified in is for the domain level log file/path.  I just recently ran into this while running ItMiiDynamicUpdate integration test and so we don't generate the override for this sparse model processing.  Unfortunately, I don't see the domain name set as an environment variable (or I missed it) when running introspection and the current introspectDomain.py gets the domain name from the domain config.xml.  Any suggestion for an alternative place to get the domain name?

Reply:
For the update case, I think the operator pod may be providing the domain name in the DOMAIN_NAME env var, as I think the Operator sets the env var when re-running the introspector (it gets the information from the 'topology' that is output from the first introspectdor run).  Alternatively, I think the 'topology' is available in the introspect cm, which is generated based on the output of the first run of the introspector...